carbon-design-system/carbon-addons-iot-react
# 2987 list enable existing checkboxes for multi selection

Closes #2987

**Summary**
Enables checkboxes for multi selection in the list

**Change List (commits, features, bugs, etc)**

- Added prop `isCheckboxMultiSelect ` to enable checkboxes without the dnd to the List
- Updated both the default and the virtual list content to support the new props
- Added prop `indeterminateIds` which is needed to support nested cases
- Added new story with non hierarchal checkboxes and updated the existing hierarchal checkbox story (which now supports updating the parent if needed when a child is modified). 
- fixed css to don't show mitered corners
- Updated docs
- Wrote new test and remove the non existing prop `isSelectable` from a few existing tests 

**Acceptance Test (how to verify the PR)**
1. Go to story https://deploy-preview-2988--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-checkbox-multi-selection
2. Verify that the multi selection works by clicking on the row, clicking the checkbox and using enter key when row is focused. 
3. Verify the same but with knob isVirtualList checked
4. Go to story https://deploy-preview-2988--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-checkbox-multi-selection-and-hierarchy
5 Repeat step 2-3 but also verify that child / parent selections works in BOTH directions including the indeterminate state


**Regression Test (how to make sure this PR doesn't break old functionality)**
- It should not break any existing List stories or stories based on the list



**Things to look for during review**
